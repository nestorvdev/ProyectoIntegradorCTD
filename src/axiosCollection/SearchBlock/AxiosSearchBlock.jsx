import axios from "axios";

const baseURL = "http://worldguestbooking.com.ar:8080/";

function AxiosGetAllCities(setData, setErrorMessage){
    axios.get(`${baseURL}cities/all`)
    .then((response) => {
      setData(response.data);     
    })
    .catch((error) => {
      setErrorMessage(error);
    });
}

export {AxiosGetAllCities}