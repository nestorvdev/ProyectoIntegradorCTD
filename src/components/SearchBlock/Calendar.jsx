import React, { useState } from "react";
import "react-datepicker/dist/react-datepicker.css";
import AdapterDateFns from "@mui/lab/AdapterDateFns";
import LocalizationProvider from "@mui/lab/LocalizationProvider";
import TextField from "@mui/material/TextField";
import StaticDateRangePicker from "@mui/lab/StaticDateRangePicker";
import Box from "@mui/material/Box";
import { createTheme, ThemeProvider } from "@mui/material/styles";
import Styles from "./styles.module.css";
const { format } = require("date-fns");

function Calendar(props) {
  const [value, setValue] = useState([props.startDate != null ? props.startDate : null, props.endDate != null ? props.endDate : null]);
  const startDate = new Date(value[0]);
  const endDate = new Date(value[1]);
  const startDateToString = `${format(startDate, "dd")} de ${format(
    startDate,
    "MMM"
  )} de ${format(startDate, "y")}`;
  const endDateToString = `${format(endDate, "dd")} de ${format(
    endDate,
    "MMM"
  )} de ${format(endDate, "y")}`;

  const handleSelection = (event) => {
    event.preventDefault();    
    if (endDate.setHours(0, 0, 0, 0) > startDate.setHours(0, 0, 0, 0)) {
      props.handleSelected(`${startDateToString} - ${endDateToString}`);
      sessionStorage.setItem("startDate", startDate.toDateString());
      sessionStorage.setItem("endDate", endDate.toDateString());
      props.setStartDate(startDate.toDateString());
      props.setEndDate(endDate.toDateString());
    }
  };

  const theme = createTheme({
    palette: {
      primary: {
        main: "#F0572D",
      },
    },
  });


  return (
    <div className={props.calendarState}>
      <ThemeProvider theme={theme}>
        <LocalizationProvider dateAdapter={AdapterDateFns}>
          <Box sx={{ minWidth: 380 }}>
            <StaticDateRangePicker
              displayStaticWrapperAs={props.responsiveness}
              showToolbar={false}
              value={value}
              minDate={new Date()}
              onChange={(newValue) => setValue(newValue)}
              renderInput={(startProps, endProps) => (
                <React.Fragment>
                  <TextField {...startProps} />
                  <Box sx={{ mx: 2 }}> to </Box>
                  <TextField {...endProps} />
                </React.Fragment>
              )}
            />
            <div className={Styles.containerCalendarSelectedDates}>
              <button
                className={Styles.selectedDatesButton}
                onClick={handleSelection}
              >
                Aplicar
              </button>
            </div>
          </Box>
        </LocalizationProvider>
      </ThemeProvider>
    </div>
  );
}

export default Calendar;
