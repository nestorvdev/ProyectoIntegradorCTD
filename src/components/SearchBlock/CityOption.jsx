import React from "react";
import Styles from "./styles.module.css"

function CityOption({handleCity, id, city, country}) {
  return (
    <React.Fragment>
      <h3 className={Styles.titleCity} onClick = {() => {handleCity(id)}} >{city},&#160;</h3>
      <h3 className={Styles.titleCountry} onClick = {() => {handleCity(id)}} >{country}</h3>
    </React.Fragment>
  );
}

export default CityOption;
