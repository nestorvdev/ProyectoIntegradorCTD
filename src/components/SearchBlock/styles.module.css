.container {
  align-items: center;  
  display: flex;
  justify-content: center;
  padding: 10px 0 25px 0; 
  position: relative;
  text-align: center;
  width: 100%; 
  background: linear-gradient(-45deg, var(--tertiary-color),  rgb(236, 236, 236), var(--tertiary-color), var(--tertiary-color));
  background-size: 800%;
  animation: fondo 4s ease-in-out; 
}

@keyframes fondo {
  0% { background-position: 0 50% } 
  100% { background-position: 120% 50% }   
}

.subContainer {
  align-items: center;
  display: flex;
  flex-direction: column;
  justify-content: space-between;
  padding-left: clamp(10px, 3.9%, 40px);
  padding-right: clamp(10px, 3.9%, 40px);
  width: 100%;
}

.gridContainer {  
  display: flex;
  flex-wrap: nowrap;
  justify-content: center;
  max-height: 40px;
  width: 100%;
}

form.gridContainer > div {
  margin-right: 10px;
}

.container h1 {
  color: #ffffff;
  font-size: var(--heading1);
  font-weight: var(--font-bold);
  padding-bottom: 10px;
}

.date {
  background-color: white;
  border: 0;
  border-radius: 5px;
  box-shadow: 0px 1px 5px rgba(0, 0, 0, 0.15);
  color: var(--secondary-color);
  cursor: pointer;
  font-size: var(--text2);
  font-weight: var(--font-medium);
  height: 100%;
  overflow: hidden;
  padding-left: 42px;
  text-align: left;
  white-space: nowrap;
  width: 100%;
}

.iconSelectBox {
  height: 40px;
  max-width: 420px;
  position: relative;
  width: clamp(254px, 33vw, 420px);
}

.optionBox {
  width: 100%;
}

.buttonSearch,
.selectedDatesButton {
  color: #ffffff;
  background: var(--primary-color);
  border:none;
  border-radius: 5px;
  box-shadow: 0px 2px 4px 0px rgba(0, 0, 0, 0.12);
  font-size: var(--button2);
  font-weight: var(--font-bold);
}

.buttonSearch {
  transition: all 0.3s;
  width: clamp(124px, 18vw, 206px);
}

.buttonSearch:hover {
  transform:scale(1.05);
  cursor: pointer;
}

.buttonSearch:active{
  transform:scale(0.95);}

.selectedDatesButton {
  bottom: 5%;
  font-size: var(--button1);
  font-weight: var(--font-bold);
  padding: 0.4rem 3.5rem;
  position: absolute;
  right: 3%;
  text-align: right;
}

.calendarButton {
  background-color: #ffffff;
  border-radius: 5px;
  height: 100%;
  position: relative;
}

.calendarButton::before {
  background: url("img/calendar.png") no-repeat;
  content: "";
  display: inline-block;
  height: 25px;
  left: 10px;
  position: absolute;
  top: 9px;
  width: 24px;
}

.calendarButton > div {
  position: absolute;
  top: 120%;
}

.hidden {
  display: none;
}

.visible {
  position: relative;
  z-index: 2;
}

.visibleTablet {
  position: absolute;
  right: -50%;
  top: 120%;
  z-index: 100;
}

.cityContainer {
  display: flex;
  position: relative;
}

.cityName {
  flex-direction: column;
  position: relative;
}

.optionLocationIcon {
  font-size: 20px;
  left: -6%;
  position: absolute;
  top: 20%;
}
div.iconSelectBox > div > div:first-child {
  border: 0px;
}

div.iconSelectBox > div:nth-child(2),
div.iconSelectBox > div:nth-child(2) > div:first-child {
  height: 100%;
}

.titleCity,
.titleCountry {
  font-size: var(--text2);
  font-weight: 500;
}

.titleCountry {
  color: var(--secondary-color);
}

@media all and (max-width:1200px) {
  .calendarButton > div {
    right: 0%;
  }
}

@media all and (max-width: 769px) {
  .container {
    padding: 25px 0px;
  }

  .container h1 {
    font-size: var(--heading2);
  }

  .cityIcon {
    left: -17%;
  }

  .iconCityBar {
    left: 4%;
  }
}

@media all and (max-width: 600px) {
  .gridContainer {
    flex-direction: column;
    max-height: 100%;
  }

  .iconSelectBox {
    max-width: 100%;
    width: 100%;
  }

  form.gridContainer > div {
    height: 40px;
  }

  form.gridContainer > div:nth-child(2) {
    margin: 10px 0;
  }

  .buttonSearch {
    height: 40px;
    width: 100%;
  }
}

@media all and (max-width: 415px) {
  .containerCalendarSelectedDates {
    background-color: white;
    padding: 0 1rem 1rem 0;
    text-align: right;
    width: 100%;
  }
  .selectedDatesButton {
    position: relative;
  }
}
