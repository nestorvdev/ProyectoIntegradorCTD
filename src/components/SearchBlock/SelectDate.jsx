import React, { useState, useEffect, useRef } from "react";
import Calendar from "./Calendar";
import Styles from "./styles.module.css"

function SelectDate({ setStartDate, setEndDate, startDate, endDate, borrarDate, setBorrarDate }) {
  const [calendarState, calendarSetState] = useState(Styles.hidden);
  const [responsiveState, responsiveSetState] = useState("desktop");

  const handleToggle = (event) => {
    const windowSize = window.innerWidth >= 680 ? (window.innerWidth <= 1080 ? Styles.visibleTablet : Styles.visible) : Styles.visible;
    event.preventDefault();
    calendarSetState(calendarState === Styles.hidden ? windowSize : Styles.hidden);
    responsiveSetState(window.innerWidth < 680 ? "mobile" : "desktop");
  };

  const [buttonState, buttonSetState] = useState("Check in - Check out");
  const handleSelected = (value) => {
    buttonSetState(value);
    calendarSetState(Styles.hidden);
  }

  const buttonRef = useRef();

  useEffect(() => {
    if (borrarDate) {
      console.log(buttonRef.current.Select);
      buttonRef.current.textContent = "Check in - Check out"
      setBorrarDate(false);
    }
    console.log(buttonRef.current.textContent);
  })



  return (
    <div className={Styles.calendarButton}>
      <button className={Styles.date} onClick={handleToggle} ref={buttonRef}>{buttonState}</button>
      <Calendar responsiveness={responsiveState} calendarState={calendarState} handleSelected={handleSelected} setStartDate={setStartDate} setEndDate={setEndDate} startDate={startDate} endDate={endDate} />
    </div>
  );
}

export default SelectDate;
