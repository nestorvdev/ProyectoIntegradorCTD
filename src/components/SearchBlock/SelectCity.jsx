/* eslint-disable no-unused-vars */
import React, { useState, useEffect, useRef } from "react";
import Select from 'react-select';
import CityOption from './CityOption';
import vector from './img/Vector.png'
import localizador from './img/localizador.png'
import { AxiosGetAllCities } from "../../axiosCollection/SearchBlock/AxiosSearchBlock";


function SelectCity({ handleCity, borrarCity, setBorrarCity }) {

  const [data, setData] = useState([]);
  const [errorMessage, setErrorMessage] = useState("");

  useEffect(() => {
    AxiosGetAllCities(setData, setErrorMessage)
  }, []);

  const selectInputRef = useRef();


  useEffect(() => {
    if (borrarCity) {
      console.log(selectInputRef.current.Select);
      selectInputRef.current.clearValue()
      setBorrarCity(false);
    }
  })


  console.log(borrarCity);



  /*
  setOnClear(handleClick)

  function handleClick(){
    selectInputRef.current.select.clearValue()
  }*/

  const customStyles = {
    option: () => ({
      borderBottom: 'solid 2px #F0572D',
      fontWeight: 500,
      padding: '0.5rem 2.5rem',
      position: 'relative',
      textAlign: 'left',
      width: '100%',
      zIndex: '10',
      ':after': {
        backgroundImage: `url(${vector})`,
        backgroundRepeat: 'no-repeat',
        borderRadius: '10px',
        content: '" "',
        display: 'block',
        height: '25px',
        marginRight: '8px',
        left: '12px',
        position: 'absolute',
        top: '16px',
        width: '25px',
      },
      ':hover': {
        backgroundColor: '#f0d7d1',
        cursor: 'pointer',
      },
      ':last-child': {
        borderBottom: 'none',
      }
    }),

    indicatorsContainer: () => ({
      display: 'none',
    }),

    valueContainer: () => ({
      alignItems: 'center',
      backgroundColor: '#FFFFFF',
      border: 'none',
      borderRadius: '5px',
      display: 'flex',
      height: '100%',
      margin: '0',
      overflow: 'hidden',
      padding: '0 0 0 0%',
      textAlign: 'left',
      width: '100%',
    }),

    control: (styles) => ({
      ...styles,
      backgroundColor: '#FFFFFF',
      boxShadow: '0px 1px 5px rgba(0, 0, 0, 0.15)',
      cursor: "pointer",
      display: 'inline-block',
      fontWeight: 500,
      height: '40px',
      maxWidth: '100%',
      overflow: 'hidden',
      padding: '0.6rem 1rem 0.6rem 2.5rem',
      whiteSpace: 'nowrap',
      width: '100%',
      ':before': {
        backgroundImage: `url(${localizador})`,
        backgroundRepeat: 'no-repeat',
        borderRadius: '10px',
        content: '" "',
        display: 'block',
        height: '25px',
        marginRight: '8px',
        left: '10px',
        position: 'absolute',
        top: '5px',
        width: '25px',
      },
    }),

    container: () => ({
      boxShadow: '0px 1px 5px rgb(0 0 0 / 15%)',
    }),

    placeholder: () => ({
      color: '#31363F',
      fontSize: '12pt',
      fontWeight: 500,
      width: '100%',
    }),

    singleValue: () => ({
      display: 'flex',
    })
  }

  const options = data.map((city) => {
    return {
      value: `${city.name}, ${city.country}`,
      label: <CityOption city={city.name} id={city.id} handleCity={handleCity} country={city.country} />,
    };
  })


  /*<button onClick={onClear}>Clear Value</button>*/

  return (
    <>
      <Select
        ref={selectInputRef}
        placeholder='¿A dónde vamos?'
        styles={customStyles}
        options={options}
        isSearchable
        isClearable
        getOptionValue={(option) => `${option.value}:`
        }
      />

    </>
  );
}

export default SelectCity;
