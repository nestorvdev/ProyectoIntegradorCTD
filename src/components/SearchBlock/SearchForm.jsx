import React from "react";
import SelectCity from "./SelectCity";
import SelectDate from "./SelectDate";
import Styles from "./styles.module.css"

function SearchForm({ borrarCity, setBorrarCity, borrarDate, setBorrarDate, handleSearch, handleCity, setStartDate, setEndDate, startDate, endDate }) {

  return (
    <form className={Styles.gridContainer} action="">
      <div className={`${Styles.iconSelectBox} grid-item`}>
        <SelectCity handleCity={handleCity} borrarCity={borrarCity} setBorrarCity={setBorrarCity}/>
      </div>
      <div className={`${Styles.iconSelectBox} grid-item`}>
        <SelectDate setStartDate={setStartDate} setEndDate={setEndDate} startDate={startDate} endDate={endDate} borrarDate={borrarDate} setBorrarDate={setBorrarDate}/>
      </div>
      {/*       <button onClick={(e) => {handleSearch(e)}} handleCity = {handleCity} className={`${Styles.buttonSearch} limitation`}>Buscar</button>*/}
      <button onClick={(e) => { handleSearch(e) }} className={`${Styles.buttonSearch} limitation`}>Buscar</button>
    </form>
  );
}

export default SearchForm;