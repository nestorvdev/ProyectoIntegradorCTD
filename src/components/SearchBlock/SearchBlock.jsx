import React from 'react';
import SearchForm from './SearchForm';
import Styles from "./styles.module.css"
import StylesApp from "../../App.module.css"

function SearchBlock({borrarCity, setBorrarCity, borrarDate, setBorrarDate, handleSearch, handleCity, setStartDate, setEndDate, startDate, endDate}) {
    return (
        <div className={`${Styles.container} ${StylesApp.delimiter}`}>
            <div className={`${Styles.subContainer} ${StylesApp.delimiterChild}`}>
                <h1>Buscá ofertas en hoteles, casas y mucho más</h1>
                <SearchForm borrarCity={borrarCity} setBorrarCity={setBorrarCity} borrarDate={borrarDate} setBorrarDate={setBorrarDate} handleSearch={handleSearch} handleCity = {handleCity} setStartDate={setStartDate} setEndDate={setEndDate} startDate={startDate} endDate={endDate}/>
            </div>
        </div>
    )
}

export default SearchBlock;