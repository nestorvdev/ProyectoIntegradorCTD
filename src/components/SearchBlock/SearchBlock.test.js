import React from 'react';
import {shallow, mount} from 'enzyme';
import '@testing-library/jest-dom';
import Calendar from './Calendar';
import CityOption from './CityOption';
import SearchBlock from "./SearchBlock";
import SearchForm from './SearchForm';
import SelectCity from './SelectCity';
import SelectDate from './SelectDate';

import Enzyme from 'enzyme';
import Adapter from '@wojtekmaj/enzyme-adapter-react-17';
import { createSerializer } from 'enzyme-to-json';
Enzyme.configure({ adapter: new Adapter() });
expect.addSnapshotSerializer(createSerializer({ mode: 'deep' }));

describe('Pruebas en <Calendar />', () => {
    let wrapper;
   
    beforeEach(() => {
        wrapper = shallow(<Calendar />);
    });

    it("Deberia llamar a event.preventDefault cuando se hace click en el boton de aplicar", ()=>{
        const event = { preventDefault: () => {} }
        let handleSelected= jest.fn()
        jest.spyOn(event, 'preventDefault')
        wrapper = shallow(<Calendar handleSelected={handleSelected}/>);
        wrapper.find('button').simulate('click', event)
                
        expect(event.preventDefault).toBeCalled()
    });

});

describe('Pruebas en <CityOption />', () => {  
    let wrapper = shallow(<CityOption />);

    beforeEach(() => {
        wrapper = shallow(<CityOption />);
    })

    it("Deberia mostrar <CityOption/> correctamente", () => {
        expect(wrapper).toMatchSnapshot();
    });
    it('<CityOption /> deberia renderizar la ciudad en props', () => {
        const city = 'Buenos Aires';
        const jsxCityOption = shallow(<CityOption city={city}/>);
        expect(jsxCityOption).toMatchSnapshot();
    });
    it('<CityOption /> deberia renderizar el pais en props', () => {
        const country = 'Argentina';
        const jsxCityOption = shallow(<CityOption country={country}/>);
        expect(jsxCityOption).toMatchSnapshot();
    });
    it('should render one <CityOption>', () => {
        expect(wrapper.find('h3')).toHaveLength(2);
    });

    it("Deberia llamar a handleCity cuando se hace click en la ciudad deseada", ()=>{
        let handleCity = jest.fn()
        wrapper = shallow(<CityOption handleCity={handleCity}/>);
        wrapper.find('h3.titleCity').simulate('click', handleCity)
                
        expect(handleCity).toBeCalled()
    })

    it("Deberia llamar a handleCity cuando se hace click en el pais", ()=>{
        let handleCity = jest.fn()
        wrapper = shallow(<CityOption handleCity={handleCity}/>);
        wrapper.find('h3.titleCountry').simulate('click', handleCity)
                
        expect(handleCity).toBeCalled()
    })
});

describe("Pruebas en <SearchBlock/>", ()=>{
    let wrapper;
   
    beforeEach(() => {
        wrapper = shallow(<SearchBlock />);
    });
    it("Deberia mostrar <SearchBlock/> correctamente", () => {
        expect(wrapper).toMatchSnapshot();
    });

})

describe("Pruebas en <SearchForm/>", ()=>{
    let wrapper;
   
    beforeEach(() => {
        wrapper = shallow(<SearchForm />);
    });
    it("Deberia mostrar <SearchForm/> correctamente", () => {
        expect(wrapper).toMatchSnapshot();
    });

    it("Deberia llamar a handleSearch cuando se hace click en el boton", ()=>{
        let handleSearch = jest.fn()
        wrapper = shallow(<SearchForm handleSearch={handleSearch}/>);
        wrapper.find('button').simulate('click', handleSearch)
                
        expect(handleSearch).toBeCalled()
    })
})

describe('Pruebas en <SelectCity />', () => {
    let wrapper = mount(<SelectCity />); //Se usa mount para testear el useEffect
    beforeEach(() => {
        wrapper = mount(<SelectCity />);
        
    })
    it("Deberia mostrar <SelectCity/> correctamente", () => {
        expect(wrapper).toMatchSnapshot();
    });

    it('<SelectCity /> se renderiza bien', () => {
        const placeholder='¿A dónde vamos?';
        const jsxSelectCity = shallow(<SelectCity />);
        expect(jsxSelectCity).toMatchSnapshot();
    })

});

describe('Pruebas en <SelectDate />', () => {
    let wrapper = shallow(<SelectDate />);
    beforeEach(() => {
        wrapper = shallow(<SelectDate />);
    })

    it("Deberia mostrar <SelectDate/> correctamente", () => {
        expect(wrapper).toMatchSnapshot();
    });

    it("Deberia llamar a event.preventDefault cuando se hace click en el boton", ()=>{
        const event = { preventDefault: () => {} }
        let handleToggle= jest.fn()
        jest.spyOn(event, 'preventDefault')
        wrapper = shallow(<SelectDate handleToggle={handleToggle}/>);
        wrapper.find('button').simulate('click', event)
                
        expect(event.preventDefault).toBeCalled()
    });

});