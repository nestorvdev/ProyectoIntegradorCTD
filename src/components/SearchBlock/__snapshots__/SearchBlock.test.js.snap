// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en <CityOption /> <CityOption /> deberia renderizar el pais en props 1`] = `
<Fragment>
  <h3
    className="titleCity"
    onClick={[Function]}
  >
    , 
  </h3>
  <h3
    className="titleCountry"
    onClick={[Function]}
  >
    Argentina
  </h3>
</Fragment>
`;

exports[`Pruebas en <CityOption /> <CityOption /> deberia renderizar la ciudad en props 1`] = `
<Fragment>
  <h3
    className="titleCity"
    onClick={[Function]}
  >
    Buenos Aires
    , 
  </h3>
  <h3
    className="titleCountry"
    onClick={[Function]}
  />
</Fragment>
`;

exports[`Pruebas en <CityOption /> Deberia mostrar <CityOption/> correctamente 1`] = `
<Fragment>
  <h3
    className="titleCity"
    onClick={[Function]}
  >
    , 
  </h3>
  <h3
    className="titleCountry"
    onClick={[Function]}
  />
</Fragment>
`;

exports[`Pruebas en <SearchBlock/> Deberia mostrar <SearchBlock/> correctamente 1`] = `
<div
  className="container delimiter"
>
  <div
    className="subContainer delimiterChild"
  >
    <h1>
      Buscá ofertas en hoteles, casas y mucho más
    </h1>
    <SearchForm />
  </div>
</div>
`;

exports[`Pruebas en <SearchForm/> Deberia mostrar <SearchForm/> correctamente 1`] = `
<form
  action=""
  className="gridContainer"
>
  <div
    className="iconSelectBox grid-item"
  >
    <SelectCity />
  </div>
  <div
    className="iconSelectBox grid-item"
  >
    <SelectDate />
  </div>
  <button
    className="buttonSearch limitation"
    onClick={[Function]}
  >
    Buscar
  </button>
</form>
`;

exports[`Pruebas en <SelectDate /> Deberia mostrar <SelectDate/> correctamente 1`] = `
<div
  className="calendarButton"
>
  <button
    className="date"
    onClick={[Function]}
  >
    Check in - Check out
  </button>
  <Calendar
    calendarState="hidden"
    handleSelected={[Function]}
    responsiveness="desktop"
  />
</div>
`;
